source/h2_stream.c

```diff
@@ -999,3 +1067,30 @@ struct aws_h2err aws_h2_stream_on_decoder_rst_stream(struct aws_h2_stream *strea
 
     return AWS_H2ERR_SUCCESS;
 }
+
+static int s_stream_write_data(struct aws_http_stream *stream_base, const struct aws_h2_data_write_options *options) {
+    struct aws_h2_stream *stream = AWS_CONTAINER_OF(stream_base, struct aws_h2_stream, base);
+    struct aws_h2_connection *connection = s_get_h2_connection(stream);
+    (void)stream;
+    (void)connection;
+    (void)options;
+
+    /* queue this new write into the pending write list for the stream */
+    struct aws_h2_stream_data_write *pending_write =
+        aws_mem_calloc(connection->base.alloc, 1, sizeof(struct aws_h2_stream_data_write));
+    pending_write->data_stream = options->data;
+    pending_write->on_complete = options->on_complete;
+    pending_write->user_data = options->user_data;
+
+    s_lock_synced_data(stream);
+    aws_linked_list_push_back(&stream->synced_data.pending_write_list, &pending_write->node);
+    /* If the stream is currently asleep */
+    if (stream->synced_data.waiting_for_writes) {
```

Comment:
need to check:
- using manual_write
- haven't called end_stream()
- stream is not complete <-- it might have died

maybe only accept if stream is in ACTIVE state?
if you accept in INIT state then you need to be sure the callback fires if cleanup happens without them ever having activated

if they think it's successful, they're waiting for the callback to happen